bin/update_protein_ids.py

```diff
@@ -29,54 +29,131 @@
+        # HEADERS
+        print('\t'.join(proteins_columns), file = outfile_proteins)
+        print('\t'.join(entities_proteins_columns), file = outfile_entities_proteins)
+        print('\t'.join(entities_columns), file = outfile_entities)
+        print('\t'.join(microbiomes_entities_columns), file = outfile_microbiomes_entities)
+
+        #
+        # PREDICTED PROTEINS
+        #
+        if args.predicted_proteins:
+            # Check validity of runtime arguments
+            if not args.predicted_proteins_microbiome_ids or len(args.predicted_proteins_microbiome_ids) != len(args.predicted_proteins):
+                sys.exit("An equal number of arguments has to be passed to --predicted-proteins and --predicted_proteins_microbiome_ids")
+            # Read all provided files
+            for mb_id, inpath in zip(args.predicted_proteins_microbiome_ids, args.predicted_proteins):
+                # Read and annotate proteins
+                proteins = pd.read_csv(inpath, sep='\t')
+                proteins['entity_name'] = proteins['protein_tmp_id'].map(lambda x : "_".join(x.split("_")[:-1]))
+                proteins['protein_id']  = range(next_protein_id, next_protein_id + len(proteins))
+                next_protein_id += len(proteins)
+                proteins.rename(columns={'protein_tmp_id' : 'protein_orig_id'}, inplace=True)
+
+                entities = proteins[['entity_name']].drop_duplicates()
+                entities['entity_id'] = range(next_entity_id, next_entity_id + len(entities))
+                next_entity_id += len(entities)
+                entities['microbiome_id'] = mb_id
+
+                # Write proteins
+                proteins[proteins_columns].to_csv(outfile_proteins, sep='\t', header=False, index=False)
+                # Write entities_proteins
+                proteins.merge(entities)[['entity_id', 'protein_id']].to_csv(outfile_entities_proteins, sep='\t', header=False, index=False)
+                # Write entities
+                entities[entities_columns].to_csv(outfile_entities, sep='\t', header=False, index=False)
+                # Write microbiomes - entities
+                entities[microbiomes_entities_columns].to_csv(outfile_microbiomes_entities, sep='\t', index=False, header=False)
+
+
+        #
+        # ENTREZ PROTEINS
+        #
+        if args.entrez_proteins:
+            # Check validity of runtime arguments
+            if not args.entrez_proteins_assemblies or not args.entrez_proteins_microbiomes:
+                sys.exit("The --entrez-* flags have to be specified together")
+            # Read proteins and associations
+            proteins = pd.read_csv(args.entrez_proteins, sep='\t')
+            proteins_assemblies = pd.read_csv(args.entrez_proteins_assemblies, '\t')\
+                    .rename(columns={
+                        'assembly_id' : 'entity_name',
+                        })
+            proteins_microbiomes   = pd.read_csv(args.entrez_proteins_microbiomes, '\t')
```

Comment:
ok, it is somehow confusing that we get rid rid of the `proteins_microbiomes` table, but still use this as an intermediate file. Could we adjust this somehow to the new format? e.g. store `microbiome_entities_tmp` using `entity_name`, i.e. `assembly_id` and then here just update it with `entity_id`?